File: python/e2.py
"""
Find the sum of even-numbered fibonacci terms below four million.
"""
from fibonaccigen import fibonacci

def Euler_2(n = 4e+6):
    #verified, integrated fibonacci() into this
    #remember f carries the value of f_prev[-1] in the while loop!
    fb = fibonacci()
    f = 0
    result = 0 #sum(f for f in fb if f % 2 == 0)

    while (f < n):
        f = next(fb)
        #test for even and sum
        if (f % 2) == 0:
            result += f
    return result

def test_2():
    assert Euler_2() == 4613732

if __name__ == "__main__":
    print(Euler_2())

File: c/e2.c
/*
Find the sum of even-numbered fibonacci terms below four million.
*/
#include <stdlib.h>
#include <stdio.h>
#define MAX_NUM 4000000

int euler_2() {
    // moved OUT of fibonaccigen.  There's not much difficult code,
    // and there is a modification that's difficult to patch in efficiently.
    int sum = 0, seq[3] = {0, 1, NULL}, n = MAX_NUM;
    while (seq[1] < n) {
        seq[2] = seq[1] + seq[0];
        seq[0] = seq[1];
        seq[1] = seq[2];
        if (seq[1] % 2 == 0) {
            sum += seq[1];
        }
    }
    return sum;
}

int main() {
    printf ("%d\n", euler_2());
    return 0;
}